src/assets/scss/_variables.scss

```diff
@@ -0,0 +1,22 @@
+:root {
+  --fw-light: 300;
+  --fw-normal: 600;
+  --fw-bold: 800;
+}
+
+:root.dark-theme {
+  --background-color: hsl(207, 26%, 17%);
+  --text-color: hsl(0, 0%, 100%);
+  --icon-fill: hsl(0, 0%, 100%);
+  --header-color: hsl(209, 23%, 22%);
+  --shadow: hsla(0, 0%, 96%, .2) 0 0 8px;
+}
+
+:root:not(.dark-theme) {
```

Comment:
nie musisz tutaj bawić się :not - ustaw sobie te style jako "domyślne" w root, a potem tylko toggluj, jak dodasz klasę, dodatkowo zrobiłbym sobie zmienne pomocnicze. Dodatkowo nie zmienialbym zmiennych globalnych raczej dodał osobne style dla wartości light i dla wartości dark, by je potem togglować w body (nie w :root) w zależności od trybu


 background-color: var(--body-bg-color);
}

Reply:
traktuje zmienne w :roocie jak stałe w JS, którymi nie manipulujesz w dużym skrócie